app/src/main/java/ca/klapstein/baudit/data/CareProvider.java

```diff
@@ -8,7 +8,27 @@
 
     private PatientTreeSet assignedPatientTreeSet;
 
-    public PatientTreeSet getAssignedPatientTreeSet() {
-        return assignedPatientTreeSet;
+    public CareProvider(Username username, Password password, ContactInfo contactInfo){
+        super(username, password, contactInfo);
+        this.assignedPatientTreeSet = new PatientTreeSet();
     }
+
+    public ProblemTreeSet getProblemTreeSet(Patient patient) {
+        if (this.getAssignedPatientTreeSet().contains(patient))
+            return patient.getProblemTreeSet();
+        else{
+            throw new IllegalArgumentException("Patient not assigned to care provider");
+        }
+    }
+    public RecordTreeSet getRecordTreeSet(Patient patient, Problem problem) {
```

Comment:
Why is this needed.

We can already access the AssignedPatientTreeSet which contains `Patient`s. So we can get an Individual `Patient` through a `TreeSet`'s public accessors.

`Patient` have `ProblemTreeSet`s which has a getters method. So we can obtain a `Patient`'s `Problem`'s.

`Problem`s have a `RecordTreeSet` which has a getters method. So we can obtain a `Patient`'s `Problem`'s `Record`s.

Each of these objects can be boiled down to successfully get sub data contained.

This method ruins this architecture by providing a quick and dirty way to access this data at the top. 
I would remove `getProblemTreeSet` and `getRecordTreeSet`.